billing/views.py

```diff
@@ -130,3 +136,13 @@ def get_object(self):
                 .get(owner=self.request.user)
         except Account.DoesNotExist:
             raise Http404('No Account matches the given query.')
+
+
+@permission_classes([permissions.IsAuthenticated])
+@api_view(['POST'])
+def pay_debt(request):
+    account = request.user.billing_account
+    assign_funds_to_account_pending_invoices(account_id=account.id)
+    charge_pending_invoices(account_id=account.id)
+    status = HTTP_400_BAD_REQUEST if account.delinquent else HTTP_200_OK
```

Comment:
The request was correct, it's only the payment that failed, so bad_request is not the right status code IMHO

Reply:
Changed